### PR TITLE
Add bazel coverage support

### PR DESCRIPTION
**Pull request by Lorite**
Fixes issue #448. I more or less followed the comments in the issue from @achew22. Then I added code to some other parts that were required. Let me know if I missed something else.
Apart from that, I also added a couple of very simple tests for the test and coverage commands to internal/e2e/simple/simple_test.go.

### CODE REVIEW

File: internal/e2e/simple/simple_test.go [hunk 1]
```diff
@@ -54,6 +64,10 @@ sh_binary(
 )
 -- subdir/subdir.sh --
 printf "Hello subdir!"
+-- test_simple_test_failing.sh --
+echo "1"
+-- test_simple_test_passing.sh --
+echo "0"
```
Comment (achew22) [inline internal/e2e/simple/simple_test.go:70]: Is there any way to have this test print out something to indicate that it is running (or not running) in coverage mode?

**Review by achew22 — CHANGES_REQUESTED** [review]

File: internal/e2e/simple/simple_test.go (hunk 1 above)
Comment (Lorite) [reply internal/e2e/simple/simple_test.go:70]: Not sure. We would probably need a coverage report generator... Something like:
`bazel coverage --combined_report=lcov --coverage_report_generator=<an sh_library rule> <a cc_test rule>`
or maybe some other bazel targets to be able to add these arguments to `bazel coverage`:
`--coverage_output_generator (a build target label; default: "@bazel_tools//tools/test:lcov_merger")
  --coverage_report_generator (a build target label; default: "@bazel_tools//tools/test:coverage_report_generator")
  --coverage_support (a build target label; default: "@bazel_tools//tools/test:coverage_support")
  --[no]collect_code_coverage (a boolean; default: "false")`

Comment (Lorite) [reply internal/e2e/simple/simple_test.go]: Let me know if you know a better approach.
I tested this implementation on a real project and it works perfectly.

File: internal/e2e/ibazel.go [hunk 2]
```diff
@@ -74,6 +74,34 @@ func (i *IBazelTester) Test(bazelArgs []string, targets ...string) {
 	i.cmd = exec.Command(ibazelPath, args...)
 	i.t.Logf("ibazel invoked as: %s", strings.Join(i.cmd.Args, " "))
 
+	// print hello world
+	i.t.Logf("Hello World")
```
Comment (achew22) [inline internal/e2e/ibazel.go:78]: Do we need this?

File: internal/e2e/simple/simple_test.go [hunk 3]
```diff
@@ -54,6 +64,12 @@ sh_binary(
 )
 -- subdir/subdir.sh --
 printf "Hello subdir!"
+-- test_simple_test_failing.sh --
+echo "I'm a failing test."
+exit 1
+-- test_simple_test_passing.sh --
+echo "I'm a passing test."
```
Comment (achew22) [inline internal/e2e/simple/simple_test.go:71]: if you invoke `env` in here are there any environment variables that differ between a `test` run and a `coverage` run?

File: internal/e2e/ibazel.go (hunk 2 above)
Comment (Lorite) [reply internal/e2e/ibazel.go]: My bad. I removed it.

File: internal/e2e/simple/simple_test.go (hunk 3 above)
Comment (Lorite) [reply internal/e2e/simple/simple_test.go:71]: I tried with printenv in both tests and all the environment variables were the same. I can't find a better way to test it than what I did.

**Comment by Lorite** [comment]
Sorry for taking so long to fix the PR. I was busy with other stuff :/

File: internal/e2e/simple/simple_test.go (hunk 3 above)
Comment (achew22) [reply internal/e2e/simple/simple_test.go:71]: I think this is as good as it is going to get

**Review by achew22 — APPROVED** [review]

**Pull request closed without merging**